examples/org.eclipse.swt.snippets/src/org/eclipse/swt/snippets/Snippet380.java

```diff
@@ -0,0 +1,63 @@
+/*******************************************************************************
+ * Copyright (c) 2022 Christoph Läubrich and others.
+ *
+ * This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License 2.0
+ * which accompanies this distribution, and is available at
+ * https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ *     Christoph Läubrich - initial API and implementation
+ *******************************************************************************/
+package org.eclipse.swt.snippets;
+
+import java.text.*;
+import java.util.concurrent.*;
+
+import org.eclipse.swt.*;
+import org.eclipse.swt.events.*;
+import org.eclipse.swt.layout.*;
+import org.eclipse.swt.widgets.*;
+
+public class Snippet380 {
+	public static void main(String[] args) {
+		Display display = new Display();
+		Shell shell = new Shell(display);
+		shell.setSize(400, 100);
+		shell.setLayout(new BorderLayout());
+		shell.setText("Snippet 380 (Executor)");
+		Button button = new Button(shell, SWT.PUSH);
+		button.setLayoutData(new BorderData(SWT.BOTTOM));
+		Label label = new Label(shell, SWT.NONE);
+		label.setLayoutData(new BorderData(SWT.CENTER));
+		button.setText("Calculate");
+		button.addSelectionListener(SelectionListener.widgetSelectedAdapter(e -> {
+			button.setEnabled(false);
+			button.setText("Calculating...");
+			CompletableFuture.supplyAsync(() -> {
```

Comment:
looks like this example is guaranteed to run in Display thread. What is the usecase for a thread where we do not know where we are?

Reply:
The future computes the value async and then you are not in the ui thread anymore. so when the computation is done you need to re-enter the UI thread (the button handler is no longer executing at that time).

Reply:
so you could also write something like 
`.thenAcceptAsync(t->display.runAsync(label.setText(t)))`
i do not see the benefit of the new executor.
Note that both could run into disposed which is wrong and a common error

Reply:
This creates a new "object" on each call, beside that I need to know the Display beforehand, even though I have it in this case I sometimes do not. beside that is much more readable instead of repeating the same code sequence over and over again. Also as noted before display.runAsync is not always suitable as you can easily deadlock then.

> Note that both could run into disposed which is wrong and a common error

Well this is a simplified example. as described before, disposal handling is not my intention here.